src/librustc/middle/dataflow.rs

```diff
@@ -181,7 +181,7 @@ fn build_local_id_to_index(body: Option<&hir::Body>,
 
     cfg.graph.each_node(|node_idx, node| {
         if let cfg::CFGNodeData::AST(id) = node.data {
-            index.entry(id).or_insert(vec![]).push(node_idx);
+            index.entry(id).or_insert(Vec::with_capacity(1)).push(node_idx);
```

Comment:
We should probably make the allocation lazy with `.or_insert_with(|| Vec::with_capacity(1))`